src/Data/Nat/DivMod.agda

```diff
@@ -274,6 +400,58 @@ m*n/m!≡n/[m∸1]! : ∀ m n .{{_ : NonZero m}} →
                  (m * n / m !) {{m !≢0}}  ≡ (n / (pred m) !) {{pred m !≢0}}
 m*n/m!≡n/[m∸1]! (suc m) n = m*n/m*o≡n/o (suc m) n (m !) {{m !≢0}} {{suc m !≢0}}
 
+m%[n*o]/o≡m/o%n : ∀ m n o .⦃ _ : NonZero n ⦄ .⦃ _ : NonZero o ⦄
+                  ⦃ _ : NonZero (n * o) ⦄ → m % (n * o) / o ≡ m / o % n
+m%[n*o]/o≡m/o%n m n o ⦃ _ ⦄ ⦃ _ ⦄ ⦃ n*o≢0 ⦄ = begin-equality
+  m % (n * o) / o                   ≡⟨ /-congˡ (m%n≡m∸m/n*n m (n * o)) ⟩
+  (m ∸ (m / (n * o) * (n * o))) / o ≡˘⟨ cong (λ # → (m ∸ #) / o) (*-assoc (m / (n * o)) n o) ⟩
+  (m ∸ (m / (n * o) * n * o)) / o   ≡⟨ [m∸n*o]/o≡m/o∸n m (m / (n * o) * n) o ⟩
+  m / o ∸ m / (n * o) * n           ≡⟨ cong (λ # → m / o ∸ # * n) (/-congʳ (*-comm n o)) ⟩
+  m / o ∸ m / (o * n) * n           ≡˘⟨ cong (λ # → m / o ∸ # * n) (m/n/o≡m/[n*o] m o n ) ⟩
+  m / o ∸ m / o / n * n             ≡˘⟨ m%n≡m∸m/n*n (m / o) n ⟩
+  m / o % n                         ∎
+  where instance o*n≢0 = subst NonZero (*-comm n o) n*o≢0
+
+m%n*o≡m*o%[n*o] : ∀ m n o .⦃ _ : NonZero n ⦄ ⦃ _ : NonZero (n * o) ⦄ →
+                  m % n * o ≡ m * o % (n * o)
+m%n*o≡m*o%[n*o] m n o ⦃ _ ⦄ ⦃ n*o≢0 ⦄ = begin-equality
+  m % n * o                         ≡⟨ cong (_* o) (m%n≡m∸m/n*n m n) ⟩
+  (m ∸ m / n * n) * o               ≡⟨ *-distribʳ-∸ o m (m / n * n) ⟩
+  m * o ∸ m / n * n * o             ≡˘⟨ cong (λ # → m * o ∸ # * n * o) (m*n/o*n≡m/o m o n) ⟩
+  m * o ∸ m * o / (n * o) * n * o   ≡⟨ cong (m * o ∸_) (*-assoc (m * o / (n * o)) n o) ⟩
+  m * o ∸ m * o / (n * o) * (n * o) ≡˘⟨ m%n≡m∸m/n*n (m * o) (n * o) ⟩
+  m * o % (n * o)                   ∎
+
+[m*n+o]%[p*n]≡[m*n]%[p*n]+o : ∀ m {n o} p ⦃ _ : NonZero (p * n) ⦄ → o < n →
+                              (m * n + o) % (p * n) ≡ (m * n) % (p * n) + o
+[m*n+o]%[p*n]≡[m*n]%[p*n]+o m {n} {o} p@(suc p∸1) ⦃ p*n≢0 ⦄ o<n = begin-equality
+  (mn + o) % pn           ≡⟨ %-distribˡ-+ mn o pn ⟩
+  (mn % pn + o % pn) % pn ≡⟨ cong (λ # → (mn % pn + #) % pn) (m<n⇒m%n≡m lem₁) ⟩
+  (mn % pn + o) % pn      ≡⟨ m<n⇒m%n≡m lem₃ ⟩
+  mn % pn + o             ∎
+  where
+  mn = m * n
+  pn = p * n
+
+  lem₁ : o < pn
```

Comment:
`m<n⇒m<o*n` is probably worthy of it's own lemma in `Data.Nat.Properties`?

Reply:
Yes, good point! For addition, we have things like ` ≤-stepsʳ` and friends. I'll add proofs for multiplication.

Reply:
Let's not call it `steps` though, I really hate that name!

Reply:
Yes. `m<n⇒m<o*n` is definitely more descriptive.